crates/executor/src/transaction.rs

```diff
@@ -616,41 +622,33 @@ fn get_max_l2_gas_amount_covered_by_balance(
 
     match tx {
         Transaction::Account(account_transaction) => {
-            match account_transaction.tx {
-                starknet_api::executable_transaction::AccountTransaction::Declare(_)
-                | starknet_api::executable_transaction::AccountTransaction::Invoke(_) => {
-                    let fee_token_address = block_context
-                        .chain_info()
-                        .fee_token_address(&account_transaction.fee_type());
-                    let balance = state.get_fee_token_balance(
-                        account_transaction.sender_address(),
-                        fee_token_address,
-                    )?;
-                    let balance = (balance.1.to_biguint() << 128) + balance.0.to_biguint();
-
-                    tracing::trace!(%balance, "Balance");
-
-                    if balance > max_possible_fee_without_l2_gas.0.into() {
-                        // The maximum amount of L2 gas that can be bought with the balance.
-                        let max_amount = (balance - max_possible_fee_without_l2_gas.0)
-                            / initial_resource_bounds
-                                .l2_gas
-                                .max_price_per_unit
-                                .0
-                                .max(1u64.into());
-                        Ok(u64::try_from(max_amount).unwrap_or(u64::MAX).into())
-                    } else {
-                        // Balance is less than committed L1 gas and L1 data gas, tx will fail
-                        // anyway. Let it pass through here so that
-                        // execution returns a detailed error.
-                        Ok(GasAmount::ZERO)
-                    }
-                }
-                starknet_api::executable_transaction::AccountTransaction::DeployAccount(_) => {
-                    Ok(block_context
-                        .versioned_constants()
-                        .initial_gas_no_user_l2_bound())
-                }
+            let fee_token_address = block_context
+                .chain_info()
+                .fee_token_address(&account_transaction.fee_type());
+            let balance = state
+                .get_fee_token_balance(account_transaction.sender_address(), fee_token_address)?;
+            let balance = (balance.1.to_biguint() << 128) + balance.0.to_biguint();
```

Comment:
Let's add a dedicated conversion function.

Reply:
Should be done in https://github.com/eqlabs/pathfinder/pull/3056/commits/963840d2ff641a980ae868ec55797ae8221686d0.